src/jailer/src/cgroup.rs

```diff
@@ -1,25 +1,224 @@
 // Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 // SPDX-License-Identifier: Apache-2.0
 
+use std::collections::hash_map::Entry::{Occupied, Vacant};
+use std::collections::HashMap;
 use std::ffi::OsStr;
 use std::fs::{self, File};
 use std::io::{BufRead, BufReader};
-use std::path::PathBuf;
+use std::path::{Path, PathBuf};
 use std::process;
+use std::str::FromStr;
 
 use regex::Regex;
 
 use crate::{readln_special, writeln_special, Error, Result};
 
-const PROC_MOUNTS: &str = "/proc/mounts";
+const PROC_MOUNTS: &str = if cfg!(test) {
+    "/tmp/firecracker/test/jailer/proc/mounts"
+} else {
+    "/proc/mounts"
+};
+
 const NODE_TO_CPULIST: &str = "/sys/devices/system/node/node"; // This constant should be removed once the `--node` argument is removed.
 
-pub struct Cgroup {
+// Enum identify cgroup version to be used
+#[derive(PartialEq, Copy, Clone)]
+pub enum CgroupVersion {
```

Comment:
Does an u8 work here instead of the enum ?

Reply:
It can work but enum would be much safer IMO because it forces handling of both variants when needed.
Also it reduces the need for error handling for values which are not 1 or 2.

Reply:
I agree there is benefit generally when using enums - like making it easy to change the value in the future (and not the code), easier to read code (when dealing with 0x123131 constants) and avoid bugs due to typos. Also, you still validate non 1|2 values in FromStr trait.

I don't have a strong preference on this, but I do have related to increasing LoC :)

Reply:
Replaced with u8.